internal/venice-common/src/main/java/com/linkedin/venice/controller/transport/ControllerGrpcTransportClient.java

```diff
@@ -0,0 +1,103 @@
+package com.linkedin.venice.controller.transport;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.venice.client.exceptions.VeniceClientException;
+import com.linkedin.venice.controller.converters.GrpcConvertersRegistry;
+import com.linkedin.venice.controller.converters.RequestConverter;
+import com.linkedin.venice.controller.converters.ResponseConverter;
+import com.linkedin.venice.controller.requests.ControllerRequest;
+import com.linkedin.venice.controllerapi.ControllerResponse;
+import com.linkedin.venice.grpc.GrpcUtils;
+import com.linkedin.venice.protocols.VeniceControllerGrpcServiceGrpc;
+import com.linkedin.venice.security.SSLFactory;
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.TlsChannelCredentials;
+import java.lang.reflect.Method;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+
+/**
+ * Transport client implementation to interact with gRPC server. The current implementation keeps the lifecycle
+ * of the client at a request granularity as opposed to a longer client due to its usage pattern across the codebase.
+ * Consider expanding the lifecycle to per component or per JVM as we sit with evolution.
+ */
+public class ControllerGrpcTransportClient implements AutoCloseable {
+  private static final long SHUTDOWN_TIMEOUT = 10000L;
+  private final ManagedChannel channel;
+
+  private final GrpcConvertersRegistry converterRegistry;
+
+  public ControllerGrpcTransportClient(String server) {
+    this(server, InsecureChannelCredentials.create());
+  }
+
+  public ControllerGrpcTransportClient(String server, SSLFactory sslFactory) {
+    this(server, buildChannelCredentials(sslFactory));
+  }
+
+  private ControllerGrpcTransportClient(String server, ChannelCredentials channelCredentials) {
+    this.channel = Grpc.newChannelBuilder(server, channelCredentials).build();
+    this.converterRegistry = new GrpcConvertersRegistry();
+  }
+
+  public <REQ extends ControllerRequest, RES extends ControllerResponse> RES request(REQ request)
+      throws ExecutionException, TimeoutException {
+
+    GrpcRoute grpcRoute = RouteUtils.getGrpcRouteFor(request.getRoute());
+
+    try {
+      VeniceControllerGrpcServiceGrpc.VeniceControllerGrpcServiceBlockingStub stub = buildStub();
```

Comment:
We've merged several PRs introducing server-side changes. In the latest updates, we've split APIs into separate services (e.g., StoreGrpcService and ClusterAdminOpsGrpcService). Going forward, additional services will be introduced to ensure clear separation of responsibilities and avoid overloading a single service.

With these changes, the client-side can no longer rely on a single stub. Instead, service-specific stubs will need to be used for each API. However, we can still reuse the same channels for these different APIs.